helix-core/src/main/java/org/apache/helix/task/TaskStateModel.java

```diff
@@ -283,6 +297,63 @@ public void reset() {
     }
   }
 
+  /**
+   * Loads className using classLoader
+   * @param classLoader
+   * @param className
+   * @return Class className loaded by classLoader
+   */
+  private Class loadClass(URLClassLoader classLoader, String className) {
+    try {
+      return classLoader.loadClass(className);
+    } catch (ClassNotFoundException e) {
+      LOG.error("Failed to load Task class " + className + " for new task in instance " + _manager
+          .getInstanceName() + " in cluster " + _manager.getClusterName() + ".");
+      throw new IllegalStateException("Null TaskFactory for task");
+    }
+  }
+
+  /**
+   * Loads Task and TaskFactory classes for command input from
+   * a JAR file, and registers the TaskFactory in _taskFactoryRegistry.
+   * @param command The command indicating what task to be loaded
+   */
+  private void loadNewTask(String command) {
+    // Read ZNRecord containing task definition information.
+    ZNRecord taskConfig = _manager.getHelixDataAccessor().getBaseDataAccessor()
+        .get(TASK_PATH + "/" + command, null, 0);
```

Comment:
I would suggestion to make this feature optional. If there is no folder created for "TASK_DEFINITON", let's skip the steps. Otherwise, it could bring potential risks.

Reply:
Thank you for your comments, Junkai. So do you think something like this at the beginning of the method would do?

```suggestion
    // If the path for dynamic tasks doesn't exist, skip loading the task
    if (!_manager.getHelixDataAccessor().getBaseDataAccessor().exists(TaskConstants.TASK_PATH, 0)) {
      return;
    }
    ZNRecord taskConfig = _manager.getHelixDataAccessor().getBaseDataAccessor()
        .get(TASK_PATH + "/" + command, null, 0);
```